test/fixtures_s3_test.go

```diff
@@ -0,0 +1,165 @@
+		s.Fail("failed to boot mocks: %s", err.Error())
+
+		return
+	}
+
+	s.mocks = mocks
+	s.client = s.mocks.ProvideS3Client("s3")
+	s.logger = mon.NewLogger()
+
+	config := cfg.New()
+	err = config.Option(
+		cfg.WithConfigFile("test_configs/config.s3.test.yml", "yml"),
+		cfg.WithConfigFile("test_configs/config.fixtures_s3.test.yml", "yml"),
+		cfg.WithConfigMap(s.s3Config()),
+	)
+	s.NoError(err)
+	s.loader = fixtures.NewFixtureLoader(config, s.logger)
+}
+
+func (s *FixturesS3Suite) TearDownSuite() {
+	s.mocks.Shutdown()
+}
+
+func TestFixturesS3Suite(t *testing.T) {
+	suite.Run(t, new(FixturesS3Suite))
+}
+
+func (s FixturesS3Suite) TestS3() {
+	err := s.loader.Load(s3DisabledPurgeFixtures())
+	s.NoError(err)
+
+	goi := &aws_s3.GetObjectInput{
+		Bucket: aws.String(BucketName),
+		Key:    aws.String(fmt.Sprintf("%s/HappyFace.jpeg", BasePath)),
+	}
+	goo, err := s.client.GetObject(goi)
+
+	s.NoError(err)
+	s.Equal(int64(11404), *goo.ContentLength)
+
+	goi = &aws_s3.GetObjectInput{
+		Bucket: aws.String(BucketName),
+		Key:    aws.String(fmt.Sprintf("%s/subDir/homer_simpson.gif", BasePath)),
+	}
+	goo, err = s.client.GetObject(goi)
+
+	s.NoError(err)
+	s.Equal(int64(54736), *goo.ContentLength)
+
+	goi = &aws_s3.GetObjectInput{
+		Bucket: aws.String(BucketName),
+		Key:    aws.String(fmt.Sprintf("%s/subDir/nyan_cat.gif", BasePath)),
+	}
+	goo, err = s.client.GetObject(goi)
+
+	s.NoError(err)
+	s.Equal(int64(28092), *goo.ContentLength)
+}
+
+func (s FixturesS3Suite) TestS3WithPurge() {
+	// Create something to purge first.
+	err := s.loader.Load(s3DisabledPurgeFixtures())
+	s.NoError(err)
+
+	goi := &aws_s3.GetObjectInput{
+		Bucket: aws.String(BucketName),
+		Key:    aws.String(fmt.Sprintf("%s/subDir/homer_simpson.gif", BasePath)),
+	}
+	goo, err := s.client.GetObject(goi)
+
+	s.NoError(err)
+	s.Equal(int64(54736), *goo.ContentLength)
+
+	err = s.loader.Load(s3EnabledPurgeFixtures())
+	s.NoError(err)
+
+	goi = &aws_s3.GetObjectInput{
+		Bucket: aws.String(BucketName),
+		Key:    aws.String(fmt.Sprintf("%s/HappyFace.jpeg", BasePathPurge)),
+	}
+	goo, err = s.client.GetObject(goi)
+
+	s.NoError(err)
+	s.Equal(int64(11404), *goo.ContentLength)
+
+	goi = &aws_s3.GetObjectInput{
+		Bucket: aws.String(BucketName),
+		Key:    aws.String(fmt.Sprintf("%s/HappyFace.jpeg", BasePath)),
+	}
+	goo, err = s.client.GetObject(goi)
+
+	s.Nil(goo.Body)
+}
+
+func s3DisabledPurgeFixtures() []*fixtures.FixtureSet {
+	return []*fixtures.FixtureSet{
+		{
+			Enabled: true,
+			Writer: fixtures.S3FixtureWriterFactory(&s3.Settings{
+				BucketName: BucketName,
+				BasePath: BasePath,
+			}),
+			Fixtures: nil,
+		},
+	}
+}
+
+func s3EnabledPurgeFixtures() []*fixtures.FixtureSet {
+	return []*fixtures.FixtureSet{
+		{
+			Enabled: true,
+			Purge: true,
+			Writer: fixtures.S3FixtureWriterFactory(&s3.Settings{
+				BucketName: BucketName,
+				BasePath: BasePathPurge,
+			}),
+			Fixtures: nil,
+		},
+	}
+}
+
+func (s FixturesS3Suite) s3Config() map[string]interface{} {
```

Comment:
in fact this shouldn't be necessary, was it for you? The (new-style) integration tests usually replace endpoints properly. Maybe have a quick look for how it works for dynamodb or sqs and see if there is adjustments needed in the integration test execution code.

Reply:
It was necessary for me. dynamodb does it the same way. I prefer keeping it this way instead of hard coding `address:port` for tests.

Reply:
new-style integration tests are replacing endpoints properly, yes, but this is an old-style integration test, so you have to set it yourself?

Reply:
in that case: why not make it a new style integration test instead?